### PR TITLE
Dev ops working

### PR DESCRIPTION
**Pull request by BeoHawk25**
Initial Azure pipeline - DevOps-Working branch will be deleted after merge in favor of short-lived feature branches

### CODE REVIEW

**Review by jpbulloch5 — DISMISSED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#     messageType: 'content'
+#     content: | 
+#           __*Team 1 P3*__  Analysis Report
+#           ``` Build Number = $(Build.BuildNumber)
+#           Report = https://sonarcloud.io/dashboard?branch=$(Build.SourceBranch)&id=1053-August-Duet-Project-Registry_project-registry-${{ variables.microservice }}
+#           Repo = $(Build.Repository.Name)
+#           Branch = $(Build.SourceBranch)
+#           Trigger = $(Build.Reason)
+#           Status = $(Agent.JobStatus)```
+#   displayName: Notification --> Task
```
Comment (escalonn) [inline azure-pipelines.yml]: commented out code that's not ready doesn't go to an integration branch like dev, it stays in other branches until ready.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+trigger:
+- Dev
+
+pool:
+  vmImage: ubuntu-latest
+
+variables:
+  microservice: tracking-microservice
+  coveragePath: src/main/java/com/revature/registry/aspect,controller,model,repository,service
```
Comment (escalonn) [inline azure-pipelines.yml]: i don't think [this syntax](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/maven?view=azure-devops) is right. i think you need to repeat the prefix for each comma-separated item, which is
probably best done with a variable and one of the multiline string syntaxes that takes out the actual newlines.

that said, are you sure you need to be so specific about each folder here? how come?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+- task: SonarCloudPublish@1
+  inputs:
+    pollingTimeoutSec: '300'
+  displayName: Publish Analysis of ${{ variables.microservice }} --> Task
+
+# - task: 'ado-discord-webhook@1'
+#   condition: always()
+#   inputs:
+#     channelId: '856587514218545203'
+#     webhookKey: 'OmVq23OAAKWh6VQGfoiFE-DRlGwtDst19OLHedR9HaQ7eONnVdlBJlc1i8OiAUpz4hqd'
```
Comment (escalonn) [inline azure-pipelines.yml]: are you sure this value shouldn't be kept hidden from public github?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- task: SonarCloudPrepare@1
+  inputs:
+    SonarCloud: 'SonarCloud'
+    organization: '1053-august-duet-project-registry'
+    scannerMode: 'Other'
+    extraProperties: |
+      # Additional properties that will be passed to the scanner, 
+      # Put one key=value per line, example:
+      # sonar.exclusions=**/*.bin
+      sonar.projectKey=1053-August-Duet-Project-Registry_project-registry-${{ variables.microservice }}
```
Comment (escalonn) [inline azure-pipelines.yml]: things that would change on the next iteration of this project, like what github org, what sonarcloud org, sonar key, etc; should all be kept in one place like variables.

**Review by escalonn — CHANGES_REQUESTED** [review]
uppercase letters should be avoided in git branch names because of case sensitivity issues which can propagate from the filesystem to git in [cases like this](https://news.ycombinator.com/item?id=23046374).

you can create new branches even with the existing ones, then change the source and target branches for this PR without needing to delete it and recreate it, then delete the uppercase-lettered branches. i recommend everyone re-clone their local repos after this or it'll probably just cause the exact issue i linked.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
+trigger:
+- Dev
```
Comment (escalonn) [inline azure-pipelines.yml]: don't we want to run this pipeline on main as well as dev?

**Comment by escalonn** [comment]
the name for this PR needs to be more descriptive, and the description should be a little more descriptive as well, of what things are implemented in the pipeline so far.

**Pull request closed without merging**